crates/bevy_animation/src/lib.rs

```diff
@@ -520,7 +603,48 @@ fn apply_animation(
                         let translation_start = keyframes[step_start];
                         let translation_end = keyframes[step_start + 1];
                         let result = translation_start.lerp(translation_end, lerp);
-                        transform.translation = transform.translation.lerp(result, weight);
+                        let translation = transform.translation.lerp(result, weight);
+                        if Some(bone_id) == root_bone_id {
+                            let root_transform = transforms.get(root).unwrap();
```

Comment:
There could be a
```rust
let Some(root_motion) = animation.root_motion else {
  unexpected!("`Some(bone_id) == root_bone_id` depends on `root_motion` being `Some`");
};
```
Then `root_motion` can be used instead of unwrapping each time.